ansible/roles/kafka/tasks/deploy.yml

```diff
@@ -25,22 +24,28 @@
 - name: (re)start kafka
   docker:
     name: kafka
-    image: "{{ docker_registry }}{{ docker_image_prefix }}/kafka:{{ docker_image_tag }}"
+    image: "ches/kafka:0.10.0.1"
     state: reloaded
     restart_policy: "{{ docker.restart.policy }}"
-    log_driver: syslog
     links:
       - "zookeeper:zookeeper"
     env:
-      "EXPOSED_HOST": "{{ inventory_hostname }}"
+      "KAFKA_ADVERTISED_HOST_NAME": "{{ inventory_hostname }}"
     ports:
       - "{{ kafka.port }}:9092"
-      - "{{ kafka.ras.port }}:8080"
 
-- name: wait until the Kafka in this host is up and running
-  uri:
-    url: "http://{{ inventory_hostname }}:{{ kafka.ras.port }}/ready"
+- name: wait until the kafka server started up
+  shell: docker logs kafka
   register: result
-  until: (result.status == 200) and (result.json.msg == 'yes')
-  retries: 36
-  delay: 5
+  until: ('[Kafka Server 0], started' in result.stdout)
+  retries: 10
+  delay: 2
+
+- name: create the active-ack topic
+  shell: "docker exec kafka bash -c 'unset JMX_PORT; kafka-topics.sh --create --topic {{ item }} --replication-factor 1 --partitions 1 --zookeeper {{ inventory_hostname }}:{{ zookeeper.port }}'"
+  with_items:
+    - completed
+
+- name: create the invoker topics
+  shell: "docker exec kafka bash -c 'unset JMX_PORT; kafka-topics.sh --create --topic invoke{{ item.0 }} --replication-factor 1 --partitions 1 --zookeeper {{ inventory_hostname }}:{{ zookeeper.port }}'"
```

Comment:
this is interesting. IIRC Kafka [creates a topic automatically](http://kafka.apache.org/documentation.html#basic_ops_add_topic) when data is first published to a non-existing topic, using default settings for replication factor and partitions.